### PR TITLE
Vs-eng-generator-interactable-lv-520

### PR DESCRIPTION
**Pull request by RyanASwanson**
https://bradleycapstone.atlassian.net/jira/software/projects/LV/boards/32/backlog?assignee=6318bb796856bdd60aa045de&selectedIssue=LV-520
Tbh the generator functionality ended up being a small part of this PR. Had to make adjustments to how attacks work to forcibly end certain attacks. The words from the task of "ends all current attacks and progress to the next story beat." Takes up a decent chunk of this entire PR. Also had to hook up the Story Manager to the game and have it spawn enemies on the correct act.
How to test
-Go to the main scene. I added an option to progress to the next beat on pressing 'U' Once you are on the third beat (When the debug displays 2, because it starts at 0) You can interact with the generator. This will progress to the next beat and spawn the lockdown attack. Then after picking up the spanner (its a grey box on a table) you can interact with the generator again to end the beat which kills any remaining enemies

### CODE REVIEW

File: Assets/GameAssets/Boss/BossScripts/Attacks/BossAttackActSystem.cs
Comment (ASLH68) [inline Assets/GameAssets/Boss/BossScripts/Attacks/BossAttackActSystem.cs:1]: add self as contributor

File: Assets/GameAssets/Boss/BossScripts/Attacks/BaseBossAttack.cs
Comment (ASLH68) [inline Assets/GameAssets/Boss/BossScripts/Attacks/BaseBossAttack.cs:1]: add self as cont

File: Assets/General/Scripts/Story/StoryManager.cs [hunk 1]
```diff
@@ -50,6 +53,28 @@ private void DefineSingleton()
         }
     }
 
+    /// <summary>
+    /// THIS IS PURELY FOR TESTING PURPOSES
+    /// IF YOU ARE NOT CODE REVIEWING THIS TASK AND I DIDNT DELETE IT YOU CAN DIRECT ALL HATE TOWARDS RYAN :D
```
Comment (ASLH68) [inline Assets/General/Scripts/Story/StoryManager.cs:58]: hate --> ryan
lol just remember to delete

Comment (ASLH68) [reply Assets/General/Scripts/Story/StoryManager.cs:58]: or you could add a #IF UNITY ENGINE or # IF DEV BUILD so we can take advantage of this for testing

**Review by ASLH68 — APPROVED** [review]

**Review by AlecatEnjoyer — APPROVED** [review]
I approved

**Pull request merged**